src/equation_systems/equation_system.C

```diff
@@ -434,30 +434,52 @@ TimeDependentEquationSystem::BuildBilinearForms(platypus::BCMap & bc_map)
   // Build and assemble bilinear forms acting on time derivatives
   for (int i = 0; i < _test_var_names.size(); i++)
   {
+
     auto test_var_name = _test_var_names.at(i);
+
     _td_blfs.Register(test_var_name,
                       std::make_shared<mfem::ParBilinearForm>(_test_pfespaces.at(i)));
-
     bc_map.ApplyIntegratedBCs(
         test_var_name, _td_blfs.GetRef(test_var_name), _test_pfespaces.at(i)->GetParMesh());
 
-    // Apply kernels
+    // Apply kernels to td_blf
     auto td_blf = _td_blfs.Get(test_var_name);
     if (_td_blf_kernels_map.Has(test_var_name))
     {
+      td_blf->SetAssemblyLevel(_assembly_level);
       auto td_blf_kernels = _td_blf_kernels_map.GetRef(test_var_name);
 
       for (auto & td_blf_kernel : td_blf_kernels)
       {
         td_blf->AddDomainIntegrator(td_blf_kernel->createIntegrator());
       }
     }
-    // Assemble bilinear form acting on only time derivatives
-    td_blf->Assemble();
-    // if implicit, add contribution from bilinear form acting on u: {
+
+    // Recover and scale integrators from blf
```

Comment:
Worth extending this comment to make it clear a) that it's applying the dt*du/dt contributions from the operator on the trial variable and b) it wouldn't be needed if an explicit timestepping scheme were used.